components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,342 @@
+import {
+  Component, Input, OnChanges, ViewChild, ElementRef, SimpleChanges
+} from '@angular/core';
+import * as d3 from 'd3';
+import {
+  DayPercentage
+} from 'app/entities/desktop/desktop.model';
+
+
+@Component({
+  selector: 'app-simple-line-graph',
+  templateUrl: './simple-line-graph.component.html',
+  styleUrls: ['./simple-line-graph.component.scss']
+})
+
+export class SimpleLineGraphComponent implements OnChanges {
+
+  constructor(
+    private chart: ElementRef
+  ) {}
+
+  @ViewChild('svg', {static: true}) svg: ElementRef;
+  @Input() data: DayPercentage[] = [];
+  @Input() width = 900;
+  @Input() height = 156; // we want a 116px height on the ticks, this minus margins
+  private locked: number = null;
+
+  get margin() {
+    return { right: 20, left: 20, top: 20, bottom: 20 };
+  }
```

Comment:
While it seems like you would just want to say `return 20`, its important to store each of these individually, so that if you change just one number - like the right margin, the math is correct across the rest of the file.